### PR TITLE
CAD-1920 staked genesis:  bulk pools support

### PR DESCRIPTION
**Pull request by deepfire**
1. Generation of staked geneses -- geneses including pools that are active immediately since blockchain start.
1. Extended with generation of bulk pool credentials files.

### CODE REVIEW

File: cardano-cli/cardano-cli.cabal [hunk 1]
```diff
@@ -103,9 +103,12 @@ library
                      , ouroboros-consensus-shelley
                      , ouroboros-network
                      , process
+                     , QuickCheck
                      , scientific
                      , shelley-spec-ledger
+                     , shelley-spec-ledger-test
```
Comment (mrBliss) [inline cardano-cli/cardano-cli.cabal:109]: We don't want to depend on testing-related libraries in library code.

Comment (deepfire) [reply cardano-cli/cardano-cli.cabal:109]: These dependencies are only due for the `Arbitrary` instance for `Shelley.PaymentCredential StandardShelley`.
..which is basically a `data` wrapper around a `Hash`.

So I perhaps should look harder.

Comment (deepfire) [reply cardano-cli/cardano-cli.cabal:109]: To clarify -- in this case I don't care about the credential this `Hash` is supposed to represent -- I only need a hash, to fill in this field, to make a completely fake entry.

Comment (mrBliss) [reply cardano-cli/cardano-cli.cabal:109]: An example from consensus you can base yourself on:
```haskell
import           Cardano.Crypto.Hash (HashAlgorithm)
import qualified Cardano.Crypto.Hash as Hash

import           Cardano.Ledger.Crypto (ADDRHASH, Crypto)

import qualified Shelley.Spec.Ledger.API as SL
import qualified Shelley.Spec.Ledger.Keys as SL (hashWithSerialiser)

mkDummyHash :: forall h a. HashAlgorithm h => Proxy h -> Int -> Hash.Hash h a
mkDummyHash _ = coerce . SL.hashWithSerialiser @h toCBOR

mkKeyHash :: forall c discriminator. Crypto c => Int -> SL.KeyHash discriminator c
mkKeyHash = SL.KeyHash . mkDummyHash (Proxy @(ADDRHASH c))
```
(The list of imports is probably incomplete)

Comment (deepfire) [reply cardano-cli/cardano-cli.cabal:109]: Thank you a lot, Thomas!

Comment (deepfire) [reply cardano-cli/cardano-cli.cabal]: Fixed, thank you, Thomas!

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 2]
```diff
@@ -326,6 +326,7 @@ module Cardano.Api.Typed (
     toByronTxOut,
     toShelleyNetwork,
     toShelleyPoolParams,
+    toShelleyStakeAddr,
```
Comment (erikd) [inline cardano-api/src/Cardano/Api/Typed.hs:329]: Commit message could be improved. I have a good idea what all the words in the commit message mean, but I am still unsure what the commit actually does.

**Review by erikd — DISMISSED** [review]

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs [hunk 3]
```diff
@@ -339,8 +672,8 @@ getCurrentTimePlus30 =
     plus30sec = addUTCTime (30 :: NominalDiffTime)
 
 
-readShelleyGenesis
-  :: FilePath
+readShelleyGenesis ::
+     FilePath
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs:676]: Why the move?

Comment (deepfire) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs:676]: This way you can grep for the definition -- by simply adding the `::`.

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 4]
```diff
@@ -863,18 +884,73 @@ pGenesisCmd =
...
+          )
+
+    pInitialSupplyDelegated :: Parser Lovelace
+    pInitialSupplyDelegated =
+      fmap (Lovelace . fromMaybe 0) $ Opt.optional $
+        Opt.option Opt.auto
+          (  Opt.long "supply-delegated"
+          <> Opt.metavar "LOVELACE"
+          <> Opt.help "The initial coin supply in Lovelace which will be evenly distributed across initial, delegating stake holders."
+          <> Opt.value (fromIntegral (0 :: Integer))
```
[the remark this answers is not in the document]
Comment (deepfire) [reply cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs]: Right, thank you @Jimbo4350!

File: cardano-api/src/Cardano/Api/Typed.hs (hunk 2 above)
Comment (deepfire) [reply cardano-api/src/Cardano/Api/Typed.hs:329]: Done.

File: cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs (hunk 3 above)
Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs:676]: Please revert this (and other instances in this PR where you have done this). All other functions in this repo do not follow this style.

Comment (Jimbo4350) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs:676]: Will green light once this is done.

Comment (deepfire) [reply cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs]: @Jimbo4350, done.

**Review by Jimbo4350 — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 5]
```diff
@@ -863,18 +884,73 @@ pGenesisCmd =
...
+          )
+
+    pInitialSupplyDelegated :: Parser Lovelace
+    pInitialSupplyDelegated =
+      fmap (Lovelace . fromMaybe 0) $ Opt.optional $
+        Opt.option Opt.auto
+          (  Opt.long "supply-delegated"
+          <> Opt.metavar "LOVELACE"
+          <> Opt.help "The initial coin supply in Lovelace which will be evenly distributed across initial, delegating stake holders."
+          <> Opt.value 0
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:935]: @deepfire Same thing here re: `Lovelace . fromMaybe 0` and `Opt.value 0`

**Pull request merged**